apps/src/StudioApp.js

```diff
@@ -394,6 +395,11 @@ StudioApp.prototype.init = function (config) {
         var nonDropletError = false;
         // are we trying to toggle from blocks to text (or the opposite)
         var fromBlocks = this.editor.currentlyUsingBlocks;
+        var event = fromBlocks ? "block-to-text" : "text-to-block";
+        firehoseClient.putRecord(
+          'analysis-events',
+          {study: 'block-and-text-transitions', event: event}
+        );
```

Comment:
This seems like the right place.

It does seem a little weird to me that we log this event using this `firehoseClient`, and then if the transition fails, a few lines later we `logToCloud` the error (which puts it in New Relic). Do we have guidelines somewhere on how to decide what the right place to log to is? Should we be moving some of the stuff that we log to NR to use this new logger?